roaster-restassured/src/main/java/hu/icellmobilsoft/roaster/restassured/response/producer/spi/AbstractConfigurableResponseProcessor.java

```diff
@@ -0,0 +1,164 @@
+import hu.icellmobilsoft.coffee.dto.exception.enums.CoffeeFaultType;
+import hu.icellmobilsoft.roaster.restassured.response.ResponseProcessor;
+import hu.icellmobilsoft.roaster.restassured.response.producer.ResponseProcessorConfig;
+import hu.icellmobilsoft.roaster.restassured.response.producer.RestProcessor;
+
+import io.restassured.http.Header;
+import io.restassured.http.Headers;
+import io.restassured.response.Response;
+import io.restassured.specification.RequestSpecification;
+import io.restassured.specification.ResponseSpecification;
+
+/**
+ * Configurable {@link ResponseProcessor}
+ *
+ * @param <RESPONSE>
+ *            response class (any type)
+ * @author martin.nagy
+ * @since 0.5.0
+ */
+public abstract class AbstractConfigurableResponseProcessor<RESPONSE> extends ResponseProcessor<RESPONSE> {
+    private static final Pattern HEADER_PATTERN = Pattern.compile("\\s*([^\\s:]+)\\s*:\\s*([^\\s:]+)\\s*");
+
+    /**
+     * Base URI config key<br>
+     * Populated by {@link #setConfig}
+     */
+    private String baseUriKey;
+
+    /**
+     * HTTP path to call. E.g.: {@literal /foo/bar} if the URL is {@literal http://localhost/foo/bar}<br>
+     * Populated by {@link #setConfig}
+     */
+    private String path;
+
+    /**
+     * Optional HTTP request headers<br>
+     * Populated by {@link #setConfig}
+     *
+     * @see RequestSpecification#headers(Headers)
+     */
+    private Headers headers;
+
+    /**
+     * Expected REST response status code<br>
+     * Populated by {@link RestProcessor#expectedStatusCode()}
+     *
+     * @see ResponseSpecification#statusCode(int)
+     */
+    private int expectedStatusCode;
+
+    /**
+     * Optional HTTP request query params<br>
+     *
+     * @see RequestSpecification#queryParams(Map)
+     */
+    private Map<String, String> queryParams;
+
+    /**
+     * Initializes the configurable values
+     *
+     * @param config
+     *            Configuration class populated with microprofile config valuest
+     * @throws BaseException
+     *             exception
+     */
+    public void setConfig(ResponseProcessorConfig config) throws BaseException {
+        if (Objects.isNull(config)) {
+            throw new BaseException(CoffeeFaultType.INVALID_INPUT, "The config is null!");
+        }
+
+        baseUriKey = config.getBaseUriKey();
+        path = config.getPath();
+
+        Optional<String[]> headersOpt = config.getHeaders();
+        headers = headersOpt.isPresent() ? parse(headersOpt.get()) : null;
+    }
+
+    private Headers parse(String[] headerStrings) throws BaseException {
+        List<Header> headerList = new ArrayList<>();
+        for (String headerString : headerStrings) {
+            Matcher matcher = HEADER_PATTERN.matcher(headerString);
+            if (!matcher.matches()) {
+                throw new BaseException(CoffeeFaultType.INVALID_INPUT, MessageFormat.format("Invalid header: [{0}]", headerString));
```

Comment:
Logba még esetleg mehetne hogy miért invalid, mert aki használja az nem biztos hogy tud erről az ellenőrzésről, nem is fogja így megtudni milyen formában kéne megadni. Mit gondoltok?

Reply:
A dokumentációban van példa rá